### PR TITLE
Sets up project framework

### PR DESCRIPTION
**Pull request by brianbruggeman**
- Ran pep8 on all files for style
- Added original authors
- Added docs with project owner name
- Move to restructured text for docs
- Added tox.ini with many environments.
- Added tests
- Added MIT License in docs
- Overhauled setup.py
- Added metadata to project
- Updates for pip
- Added licensing tests for non-permissive licenses

### CODE REVIEW

**Comment by WhyNotHugo** [comment]
I'd vote for `setuptools-scm`, rather than hand-rolling a versioning mechanism and manually editing before each tag.

Other than that, this looks pretty good.

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+.. code:: bash
+
+    pip install -e .
+
+**option 1:**
+
+.. code:: bash
+
+    pip install -e .[dev]
+    # or: python3 setup.py develop
```
Comment (mrpudn) [inline README.rst:59]: Awesome, just switch the order here:

``` bash
pycraft
# or: python3 setup.py develop
```

Comment (brianbruggeman) [inline README.rst:59]: On the versioning tag:  I just haven't found a tool that can automate exactly what I want; so I think of versioning as a discipline thing.  

I usually use git-flow, so I have a stable/release branch (master) and an unstable/integration branch (develop).  I generally use a '-dev' on the end of my version until I submit to master; this prevents inadvertently submitting things to master.  I don't have (nor do I really want) control/ownership over this particular repository, or I would have setup a git-flow branching scheme as well.

Comment (WhyNotHugo) [inline README.rst:59]: I just mentioned that setuptools-scm manages that just fine.

The problem with manually doing it is:
- Like all manual tasks that can be automated, it's error prone
- Non-stable (eg: from git) builds don't have a clear version to identify them
- It pollutes the git history with changes that merely push the version - which is redundant when you have tags.

This does not conflict with using a master/dev branch during development. This is about versioning, not workflows.

AFAIK, there are no clear downsides to using it either.

**Pull request merged**